tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/Serverless.cs

```diff
@@ -0,0 +1,90 @@
+// <copyright file="Serverless.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation
+{
+    internal static class Serverless
+    {
+        private const string ExtensionFullPath = "/opt/extensions/datadog-agent";
+        private const string FunctionEnvame = "AWS_LAMBDA_FUNCTION_NAME";
+        private const string HandlerEnvName = "_HANDLER";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Serverless));
+
+        internal static void InitIfNeeded()
+        {
+            if (IsRunningInLambda(ExtensionFullPath))
+            {
+                Log.Debug("Sending CallTarget serverless integration definitions to native library.");
+                var serverlessDefinitions = GetServerlessDefinitions();
+                NativeMethods.InitializeProfiler(GetServerlessDefinitionsId(), serverlessDefinitions);
+                foreach (var def in serverlessDefinitions)
+                {
+                    def.Dispose();
+                }
+
+                Log.Information<int>("The profiler has been initialized with {count} serverless definitions.", serverlessDefinitions.Length);
+            }
+        }
+
+        internal static bool IsRunningInLambda(string extensionPath)
+        {
+            return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(FunctionEnvame)) && File.Exists(extensionPath);
```

Comment:
`Environment.GetEnvironmentVariable` can throw a `SecurityException` ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-6.0)) , consider using https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Util/EnvironmentHelpers.cs#L63.

According to `ExtensionFullPath` this is only compatible on linux scenarios, so maybe it worth to add that OS check to avoid any throw + exception handling in case of a SecurityException while running on Windows.

Reply:
I've switched to `EnvironmentHelpers.GetEnvironmentVariable`. It seems that File.Exists cannot fail? So maybe the OS checks is not required?

Reply:
You're right!, then you only need to switch the comparisons, and have `File.Exists` first.

Reply:
✅